domaintools/results.py

```diff
@@ -140,31 +196,106 @@ def flattened(self):
         if contacts:
             for i, contact in enumerate(contacts, start=1):
                 for contact_key, contact_value in contact.items():
-                    flat[f"contact_{contact_key}_{i}"] = " | ".join(contact_value) if type(contact_value) in (list, tuple) else contact_value
+                    flat[f"contact_{contact_key}_{i}"] = (
+                        " | ".join(contact_value) if type(contact_value) in (list, tuple) else contact_value
+                    )
 
         return flat
 
 
 class FeedsResults(Results):
-    """Returns the generator for feeds results"""
+    """
+    Real Time Threat Feeds (RTTF) returns an application/ndjson stream.
+    With this we use httpx stream to process each JSON object efficiently.
+
+    Highlevel process:
+
+    httpx stream -> yield each json line -> check status code -> yield back data to client
+
+    Returns the generator object for feeds results.
+    """
+
+    latest_feeds_status_code = None
+
+    def _make_request(self) -> Generator:
+        """
+        Creates and manages the httpx stream request, yielding data line by line.
+        This is the core generator that communicates with the DT frontend API server.
+        """
+        session_info = self._get_session_params_and_headers()
+        headers = session_info.get("headers")
+        parameters = session_info.get("parameters")
+
+        try:
+            with httpx.stream(
+                "GET",
+                self.url,
+                headers=headers,
+                params=parameters,
+                verify=self.api.verify_ssl,
+                proxy=self.api.proxy_url,
+                timeout=None,
+            ) as response:
+                self.latest_feeds_status_code = response.status_code
+                # assigned the status code to `latest_feeds_status_code`
+                # started the process
+                yield {"status_ready": True}
+
+                for line in response.iter_lines():
+                    if line:
+                        yield line
+        except Exception as e:
+            self.latest_feeds_status_code = 500
+            yield {"status_ready": True, "error": str(e)}
+            raise
+
+    def _get_results(self) -> Generator:
+        try:
+            wait_for = self._wait_time()
+            feeds_generator = self._make_request()
+
+            next(feeds_generator)  # to start the generator process
+            self.setStatus(self.latest_feeds_status_code)  # set the status already
+
+            should_wait = (
+                wait_for
+                and wait_for > 0
```

Comment:
Why are you checking if it is greater than 0? If `wait_for` is 0 then it will be `False` and checked in the previous line. I don't think `wait_for` can be negative given that it is the difference of the `safe_after` and `now`.